oresat_configs/scripts/gen_xtce.py

```diff
@@ -142,247 +152,635 @@ def write_xtce(config: OreSatConfig, dir_path: str = ".") -> None:
+
+            data_type = CANOPEN_TO_XTCE_DT[obj.data_type]
+            value_descriptions = {name: value for value, name in obj.value_descriptions.items()}
+            _add_parameter_type(
+                root=para_type_set,
+                name=para_type_name,
+                data_type=data_type,
+                description=obj.description,
+                unit=obj.unit,
+                factor=obj.factor,
+                default=obj.default,
+                value_descriptions=value_descriptions,
+            )
+
+        _add_parameter(para_set, para_name, para_type_name, obj.description)
+        _add_parameter_ref(beacon_entry_list, para_name)
+
+    # beacon tails
+    _add_parameter(para_set, "crc32", "uint32_type", "beacon crc32")
+    _add_parameter_ref(beacon_entry_list, "crc32")
+
+    # OreSat0 was before oresat-configs and had different commands
+    if config.mission != Consts.ORESAT0:
+        _add_edl(config, root, cont_set, para_type_set, para_set, para_types)
+
+    tree = ET.ElementTree(root)
+    ET.indent(tree, space="  ", level=0)
+    file_name = f"{config.mission.filename()}.xtce"
+    tree.write(f"{dir_path}/{file_name}", encoding="utf-8", xml_declaration=True)
+
+
+def _add_edl(
+    config: OreSatConfig,
+    root: ET.Element,
+    cont_set: ET.Element,
+    para_type_set: ET.Element,
+    para_set: ET.Element,
+    para_types: list[str],
+):
+    cmd_meta_data = ET.SubElement(root, "CommandMetaData")
+    arg_type_set = ET.SubElement(cmd_meta_data, "ArgumentTypeSet")
+
+    # custom argument types
+    node_ids = {}
+    for name in config.od_db:
+        if config.cards[name].node_id != 0:
+            node_ids[config.cards[name].nice_name] = config.cards[name].node_id
+    _add_argument_type(
+        arg_type_set, EdlCommandField("node_id", "uint8", enums=node_ids), "node_id_type"
     )
-    bin_data_enc_size_fixed = ET.SubElement(
-        bin_data_enc_size,
-        "FixedValue",
+    opd_addrs = {}
+    for name in config.od_db:
+        if config.cards[name].opd_address != 0:
+            opd_addrs[config.cards[name].nice_name] = config.cards[name].opd_address
+    _add_argument_type(
+        arg_type_set, EdlCommandField("opd_addr", "uint8", enums=opd_addrs), "opd_addr_type"
     )
-    bin_data_enc_size_fixed.text = "128"
 
-    # hard-code the unix time type
-    para_type = ET.SubElement(
-        tm_meta_para,
-        "AbsoluteTimeParameterType",
+    arg_types = ["opd_addr_type", "node_id_type"]
+
+    _add_parameter_type(
+        para_type_set,
+        "edl_command_code_type",
+        "uint8",
+        value_descriptions={cmd.name: cmd.uid for cmd in config.edl_cmd_defs.values()},
+    )
+    _add_parameter(para_set, "edl_command_code", "edl_command_code_type")
+    res_seq_cont = ET.SubElement(
+        cont_set,
+        "SequenceContainer",
         attrib={
-            "name": "unix_time",
-            "shortDescription": "Unix coarse timestamp",
+            "name": "edl_responses",
```

Comment:
I've been trying to come up to speed on XTCE too and I agree it's pretty confusing. I've found the project [xtcetools](https://gitlab.com/dovereem/xtcetools) and also [yamcs quickstart](https://github.com/yamcs/quickstart/blob/master/src/main/yamcs/mdb/xtce.xml) have example XTCEs that include some CCSDS related setup. Also [these](https://ntrs.nasa.gov/api/citations/20090017706/downloads/20090017706.pdf) [slide decks](https://www.omgwiki.org/space/lib/exe/fetch.php?media=xtce_tutorial_2004.pdf) have some instruction of moderate help.

They all seem to do inheritance differently than what you've got here. Where you're driving it via the single top level container `edl_responses` and selecting  with `IncludeCondition` they go the other way, making heavy use of `RestrictionCriteria` on `BaseContainer`. I think it should be something like:
```mermaid
flowchart TD
    USLP-- "RestrictionCriteria\nuslp_virtual_channel_id == 0" -->edl(EDL Response)
    edl--"RestrictionCriteria\nedl_command_code == 0"-->tx_control_response
    edl--"RestrictionCriteria\nedl_command_code == 4"-->co_node_enable_response
    edl--"RestrictionCriteria\nedl_command_code == 5"-->co_node_status_response
    edl-->...
```

Reply:
I agree with the way you describe as it was the way I originally tried to implement this as it makes more sense, the issue I ran into I could only use `IncludeCondition` once in a sequence definition according to the xsd spec.

Reply:
The key thing is I wanted on root contianer for Yamcs to point to to decode edl responses.